docs/developers/applications/README.md

```diff
@@ -69,27 +72,131 @@ Here's an example for a github repo:
+
+```javascript
+export class Greeting extends Resource {
+  get() {
+    return { greeting: 'Hello, world!' };
+  }
+}
+```
+
+In your `my-app` directory run Harper:
+
+```shell
+> harperdb run . # tell Harper cli to run current directory as an application
+```
+
+This is a quick and easy way to run you application without bundining it into a component and installed.  Well, without YOU having to do it. The command takes care of that for you.
+
+Now, make an HTTP request to your runnign application.  Note that the default port that harper listens on is 9926 which can be [configured here](../../deployments/configuration.md#http):
+
+```shell
+> curl -i http://127.0.0.1:9926/Greeting
+
+HTTP/1.1 200 OK
+Vary: Accept, Accept-Encoding
+Content-Type: application/json
+...snip headers...
+
+{"greeting":"Hello, world!"}
+```
+
+### Make it Interactive
+
+Let's change who we say hello to based on the URL query string.
+
+The `get` method endpoints is passed a `query` object:
+
+```javascript
+export class Greeting extends Resource {
+ get(query) {
+   const name = query.get('name') || 'world'
+   return { greeting: `Hello, ${name}!` };
+ }
+}
+```
+
+Stop and start harper and send the request:
+
+```shell
+> curl -i http://127.0.0.1:9926/Greeting?name=Harper
+
+HTTP/1.1 200 OK
+Vary: Accept, Accept-Encoding
+Content-Type: application/json
+...snip headers...
+
+{"greeting":"Hello, Harper!"}
+```
+
+> [!NOTE]
+> 
+> Rather than restarting harper every time you update your javascript files, you can run `harperdb dev .`. The dev mode monitors files in the application and will restart the component threads when it sees an update.  We will work with this in the next section.  Some changes, such as extention updates or logging changes might still require a full stop and start.
+
 ## Creating our first Table
 
-The core of a Harper application is the database, so let's create a database table!
+Code without data to back it up tends to have limited value.  Luckily a core part of a Harper application is the database, so let's create a database table!
 
-A quick and expressive way to define a table is through a [GraphQL Schema](https://graphql.org/learn/schema). Using your editor of choice, edit the file named `schema.graphql` in the root of the application directory, `my-app`, that we created above. To create a table, we will need to add a `type` of `@table` named `Dog` (and you can remove the example table in the template):
+A quick and expressive way to define a table is through a [GraphQL Schema](https://graphql.org/learn/schema). This leverages the built in extension called, appropriately, 'graphqlSchema'.  Using your editor of choice, edit the file named `schema.graphql` in the root of the application directory, `my-app`, that we created above. To create a table, we will need to add a `type` of `@table` named `Dog` (and you can remove the example table in the template):
 
 ```graphql
-type Dog @table {
-	# properties will go here soon
+type Dogs @table {
```

Comment:
My understanding is that it is the Harper standard to use singular names for tables, not plural (I didn't define that standard, just followed it).